core/backend/src/IModelDb.ts

```diff
@@ -573,9 +573,9 @@ export abstract class IModelDb extends IModel {
     let offset: number = 0;
     let rowsToGet = limitRows ? limitRows : -1;
     do {
-      result = await this.queryRows(ecsql, bindings, { maxRowAllowed: rowsToGet, startRowOffset: offset }, quota, priority, undefined, abbreviateBlobs);
+      result = await this.queryRows(IModelHost.sessionId, ecsql, bindings, { maxRowAllowed: rowsToGet, startRowOffset: offset }, quota, priority, undefined, abbreviateBlobs);

```

Comment:
This call can *only* originate from a non-RPC caller. In that case, the session is the backend's "session". We could just pass a constant string rather than one that's unique every session. I considered that, and may that would be preferable, just so there's no confusion.

Reply:
alternatively, if we were willing to tolerate a breaking change here, i would prefer to turn all the "options" arguments into an object. Then, we could have `sessionId?: string` as an argument.

@khanaffan ?

Reply:
@khanaffan is already in the process of breaking changes here so we may be able to get away with doing that change as well...

Reply:
In our deployed backends, we already have a string in every log message that is unique to a backend's "session" - the hostname.  That's unique to each **container** (the _actual_ VM's hostname is not visible to the container), so every single backend process has a different hostname.

Maybe we could just drop `sessionId` entirely here and just use that?

Reply:
This isn't really about logging. It's a way to identify a request for "restartable" queries. For RPC it's important to identify the caller. For "single user" (agent, IPC) I don't know if this string will ever appear in a log. If so, then @calebmshafer suggested it might be helpful for the agent use case that it be different every time the agent runs.